programs/gauge/src/state.rs

```diff
@@ -174,14 +173,14 @@ impl GaugeVote {
         &mut self,
         latest_voting_epoch: u32,
     ) -> usize {
-        let current_index: usize = self.current_index.try_into().unwrap();
-        if self.vote_epochs[current_index].voting_epoch == latest_voting_epoch {
-            return current_index;
+        let max_epoch_per_gauge: u32 = u32::try_from(MAX_EPOCH_PER_GAUGE).unwrap();
+        let index = (latest_voting_epoch % max_epoch_per_gauge) as usize;
+
+        if self.vote_epochs[index].voting_epoch == latest_voting_epoch {
+            return index;
         }
-        let current_index = current_index.checked_add(1).unwrap() % MAX_EPOCH_PER_GAUGE;
-        self.current_index = u64::try_from(current_index).unwrap();
-        self.vote_epochs[current_index].voting_epoch = latest_voting_epoch;
-        self.current_index.try_into().unwrap()
+        self.vote_epochs[index].voting_epoch = latest_voting_epoch;
```

Comment:
do we need to reset other fields?

Reply:
changed: made functions that doesn't modify the `vote_epochs` to take a `&self`

Reply:
functions that modify the `vote_epochs` should reset the inner item at index